### ISSUE TITLE
Qwen2.5 Models Assistant Masks Broken

### ISSUE THREAD

**Issue by benji-cannot-code** [NovaSky-AI/SkyRL#296]
Assistant masks for Qwen2.5 models are broken. [Hugging face discussion linked](https://huggingface.co/Qwen/Qwen2.5-VL-3B-Instruct/discussions/27). This affects any code that uses `apply_chat_template` with `return_assistant_tokens_mask=True`, such as: 

```
# inside skyrl_gym_generator.py
            response_encodings = self.tokenizer.apply_chat_template(
                chat_history[initial_chat_history_length:],
                chat_template=self.custom_chat_template,
                add_generation_prompt=False,
                return_dict=True,
                return_assistant_tokens_mask=True,
                tokenize=True,
            )
            loss_mask = response_encodings["assistant_masks"]
            response_ids = response_encodings["input_ids"]
```

and 

```
# inside base.py
        response_encodings = self.tokenizer.apply_chat_template(
            all_responses,
            chat_template=chat_template_qwen3_thinking if self.cfg.generator.remove_think_tokens else chat_template,
            return_assistant_tokens_mask=True,
            add_generation_prompt=False,
            return_dict=True,
        )

        response_ids = response_encodings['input_ids']
        response_attention_mask = response_encodings['attention_mask']
        response_assistant_mask = response_encodings['assistant_masks']
```

**Comment by benji-cannot-code** [comment]
Working on a fix now
